test/core-overlay.html

```diff
@@ -0,0 +1,336 @@
+          // first, open the overlay
+          overlay.open();
+          overlay.async(function() {
+            // during the opening transition, close the overlay
+            this.close();
+            // wait for any exceptions to be thrown until the transition is done
+            this.async(function() {
+              done();
+            }, 300);
+          });
+        });
+
+        test('clicking an overlay does not close it', function(done) {
+          runAfterOpen(overlay, function() {
+            var fn = function(event) {
+              if (!event.detail.opened) {
+                assert.fail(event.detail.opened, false, 'core-overlay-open should not be fired with !detail.opened');
+              }
+            };
+            overlay.addEventListener('core-overlay-open', fn);
+            overlay.fire('click');
+            setTimeout(function() {
+              overlay.removeEventListener('core-overlay-open', fn);
+              done();
+            }, 10);
+          });
+        });
+
+        test('cancel an overlay by clicking outside', function(done) {
+          runAfterOpen(overlay, function() {
+            overlay.addEventListener('core-overlay-cancel', function() {
+              done();
+            });
+            Polymer.Base.fire.call(document, 'click');
+          });
+        });
+
+        test('cancel an overlay with esc key', function(done) {
+          runAfterOpen(overlay, function() {
+            overlay.addEventListener('core-overlay-cancel', function() {
+              done();
+            });
+            fireEvent('keydown', {
+              keyCode: Polymer.Base.eventKeyCodes.ESC_KEY
+            }, document);
+          });
+        });
+
+        test('no-cancel-on-outside-click property', function(done) {
+          overlay.setAttribute('no-cancel-on-outside-click', '');
+          assert.isTrue(overlay.noCancelOnOutsideClick, 'noCancelOnOutsideClick set from attribute');
+          runAfterOpen(overlay, function() {
+            overlay.addEventListener('core-overlay-cancel', function() {
+              assert.fail(null, null, 'core-overlay-cancel should not be fired');
+            });
+            Polymer.Base.fire.call(document, 'click');
+            setTimeout(function() {
+              done();
+            }, 10);
+          });
+        });
+
+        test('no-cancel-on-esc-key property', function(done) {
+          overlay.setAttribute('no-cancel-on-esc-key', '');
+          assert.isTrue(overlay.noCancelOnEscKey, 'noCancelOnEscKey set from attribute');
+          runAfterOpen(overlay, function() {
+            overlay.addEventListener('core-overlay-cancel', function() {
+              assert.fail(null, null, 'core-overlay-cancel should not be fired');
+            });
+            fireEvent('keydown', {
+              keyCode: Polymer.Base.eventKeyCodes.ESC_KEY
+            }, document);
+            setTimeout(function() {
+              done();
+            }, 10);
```

Comment:
I didn't have a specific reason to choose `10` but want to wait until the key handler has fired.